utils.js

```diff
@@ -1,263 +1,295 @@
+    throw new Error(messages.SHOULD_NOT_OCCUR_ERRORS);
+  } else {
+    throw new Error(messages.FALLBACK_ERROR);
   }
+};
 
-  let archiveFile = details;
-
-  //This is all for decompressing the contents of the files See: https://www.npmjs.com/package/decompress
-  const decompress = require('decompress'); //Look for alternative for getting all of the contents of all tar file types, as of now there are no good tar readers from a buffer, only local files
-  const zip = require('node-native-zip'); //Consider swapping to this in the future: https://github.com/archiverjs/node-archiver
+//function to check whether or not the url given for the fetch
+//is absolute or the fetch got bad content. 
+const fetchChecker = (url, fetchResponse) => {
+  const lowerCase = require('lower-case');
+  let fetch = '';
 
-  //Create new zip archive, and decompress the contents
-  //of the supplied tar archive. This will fail if the tar header is corrupt or not present.
-  //Supports .bz2, .gz, and .tar (decompress module doesn't support .Z and .xz/.lzma is bugged).
-  let newArchive = new zip();
-  const toConvert = decompress(archiveFile.content, {
-    inputFile: archiveFile.content,
-  });
+  //Absolute url format checker
+  if(url.indexOf('http') !== 0){
+    fetch = 'bad url';
+  }
+  // If the url given is redirected to another place, doesn't match the given url, or contains link, the given file url
+  // wasn't the url content was extracted from. So, throw the bad fetch error.
+  else if (fetchResponse && ((fetchResponse.headers.link) || lowerCase(url) !== lowerCase(fetchResponse.url) || (fetchResponse.headers['www-authenticate']))) {
+    fetch = 'bad fetch';
+  }
 
-  return toConvert.then(files => {
-    let fileCount = 0;
+  return fetch;
+};
 
-    //Loop through the file contents and add them to the new zip archive
-    //with the file names and buffers associated with them.
-    files.forEach(file => {
-      //Ignore folders in the total file count, as we
-      //do not care about indexing folders.
-      if (file.type !== 'directory') {
-        fileCount++;
-      }
+//Function to check if the total number of files
+//processed exceeds 50.
+const fileCountChecker = fileCount => {
+  if(fileCount > 50){
+    throw new Error(messages.TOO_MANY_FILES);
+  }
+  return false;
+};
 
-      //Too many files in archive, catch early and throw an error
-      if (fileCount > 50) {
-        throw new Error(messages.TOO_MANY_FILES);
-      }
 
-      newArchive.add(file.path, file.data);
-    });
-    //Turn the entire zip in a zip file buffer, and send off to the
-    //zip file handler.
-    archiveFile.content = newArchive.toBuffer();
-    return handleZip(archiveFile);
-  });
+//Function to check if the file size or total size of the files
+//fetched exceeds 100 MB. If no size can be fetched, find it from
+//the buffer first, then check.
+const fileSizeChecker = (fileSize, fileBuffer) => {
+  if(fileSize === 'null' && fileBuffer){
```

Comment:
why would fileSize be a string `'null'` ?

Reply:
Sometimes when calling fetch on a file, the file size can't be found. So in the response headers of the fetch, instead of returning 0, null, or undefined, it returns a string 'null' for some reason. Seems dumb to return 'null', but that's just how fetch works.